### DISCUSSION TITLE
The update is improperly signed and could not be validated. Please try again later or contact the app developer.

### DISCUSSION THREAD

**Discussion by haochentao** [sparkle-project/Sparkle#2252] [Q&A]
Prompt after downloading:
The update is improperly signed and could not be validated. Please try again later or contact the app developer.
help！！

**Comment by zorgiepoo** [comment]
Check the console logs. The update archive is not signed with Sparkle's EdDSA keys properly or the app bundle is not Apple code signed correctly. Did you use generate_appcast?

**Comment by haochentao** [comment]
generate_appcast  is used
Error: EdDSA signature does not match. Data of the update file being checked is different than data that has been signed, or the public key and the private key are not from the same set. (null) (URL (null))

**Comment by zorgiepoo** [comment]
Check discussion in https://github.com/sparkle-project/Sparkle/discussions/2174 as it's a similar problem. Either the EdDSA signature is incorrect or you're serving the wrong update archive. As noted there you can try to verify the signature using `sign_update` with `--verify` option. Re-running `generate_appcast` will generate new EdDSA signatures.
